tests/Builder/BuilderEncoderTest.php

```diff
@@ -337,10 +343,89 @@ public function testRedactStage(): void
         $this->assertSamePipeline($expected, $pipeline);
     }
 
+    public function testDateTimeEncoding(): void
+    {
+        $dateTimeImmutable = new DateTimeImmutable();
+        $dateTime = DateTime::createFromImmutable($dateTimeImmutable);
+        $utcDateTime = new UTCDateTime($dateTime);
+
+        $pipeline = new Pipeline(
+            Stage::match(
+                utc: $utcDateTime,
+                mutable: $dateTime,
+                immutable: $dateTimeImmutable,
+            ),
+            Stage::addFields(
+                utc: Expression::dateToString($utcDateTime),
+                mutable: Expression::dateToString($dateTime),
+                immutable: Expression::dateToString($dateTimeImmutable),
+            ),
+        );
+
+        $expected = [
+            [
+                '$match' => [
+                    'utc' => $utcDateTime,
+                    'mutable' => $utcDateTime,
+                    'immutable' => $utcDateTime,
+                ],
+            ],
+            [
+                '$addFields' => [
+                    'utc' => ['$dateToString' => ['date' => $utcDateTime]],
+                    'mutable' => ['$dateToString' => ['date' => $utcDateTime]],
+                    'immutable' => ['$dateToString' => ['date' => $utcDateTime]],
+                ],
+            ],
+        ];
+
+        $this->assertSamePipeline($expected, $pipeline);
+    }
+
+    public function testCustomEncoder(): void
+    {
+        $customEncoders = [
+            FieldPathInterface::class => new class implements Encoder {
+                use EncodeIfSupported;
+
+                public function canEncode(mixed $value): bool
+                {
+                    return $value instanceof FieldPathInterface;
+                }
+
+                public function encode(mixed $value): mixed
```

Comment:
Adding a `mixed` return type seems necessary to address the following error in some CI builds:

> Fatal error: Declaration of MongoDB\Codec\Encoder@anonymous::encode(mixed $value) must be compatible with MongoDB\Codec\EncodeIfSupported::encode(mixed $value): mixed in /data/mci/4daf5c5b734b4e3a8c1f9bfc56625d5d/src/tests/Builder/BuilderEncoderTest.php on line 396

It's not clear to me why this didn't show up in _all_ builds.

Reply:
That's right that The error report should be constant. We will have to backport this fix to v1.x of the error occurs.

We could also have used the `@template-implements Encoder<string, FieldPathInterface>` from `FieldPathEncoder`. But `mixed` is an acceptable solution in the test.